### ISSUE TITLE
SCAN leads to fatal error with RedisArray

### ISSUE THREAD

**Issue by anbuc** [phpredis/phpredis#1393]
### Expected behaviour
Script outputs the number of keys

### Actual behaviour
Script fails with `PHP Fatal error:  RedisArray::__call(): Could not find key in /home/anbuc/workspace/phpredis/test.php on line 6`

### I'm seeing this behaviour on
- OS: Debian
- Redis: redis-cli 2.8.17
- PHP: 7.1.13
- phpredis: develop (3.1.4RC1-168-g1d99787)

### Steps to reproduce, backtrace or example script
```php
<?php

$redis = new RedisArray(['localhost:6379']);
$iterator = null;
$count = 0;
while($keys = $redis->scan($iterator)) {
    $count += count($keys); // doesn't really matter
}
echo $count;
```

### I've checked
- [x] There is no similar issue from other users
- [x] Issue isn't fixed in `develop` branch

**Comment by yatsukhnenko** [comment]
It seems that we need to implement `RedisArray::scan` method similarly to `RedisArray::keys`

**Comment by akalongman** [comment]
+1

**Comment by andrerom** [comment]
On sscan getting errors like:
`Parameter 2 to Redis::sscan() expected to be a reference, value given`

On code that seems to work with Redis class:
```
$cursor = null;
while (false !== ($chunkIds = $this->redis->sScan('key', $cursor, '', self::SSCAN_LIMIT))) {
```

**Comment by akalongman** [comment]
@yatsukhnenko seems like this does not work. My test code:

```php
<?php

ini_set('error_reporting', E_ALL);
ini_set('display_errors', 1);

$servers = [
    '127.0.0.1:6379',
    '127.0.0.1:6380',
];

$redis = new RedisArray($servers);

$redis->set('key1', 'asdf');
$redis->set('key2', 'asdf');

$iterator = null;
$count = 0;
while ($keys = $redis->scan($iterator)) {
    $count += count($keys);
}

echo $count; // Shows 0

```

**Comment by yatsukhnenko** [comment]
@akalongman `$iterator` must be array here

**Comment by akalongman** [comment]
@yatsukhnenko hmm, works now. I have to update PHPStorm code completion after this will be merged.

Can we throw a warning, in this case, if $iterator type is not an array?

**Comment by yatsukhnenko** [comment]
Fixed in 5.3.7
